File: src/main/java/io/github/starwishsama/namelessbot/BotConstants.java
package io.github.starwishsama.namelessbot;

import io.github.starwishsama.namelessbot.objects.BotLocalization;
import io.github.starwishsama.namelessbot.objects.Config;
import io.github.starwishsama.namelessbot.objects.draws.ArkNightOperator;
import io.github.starwishsama.namelessbot.objects.draws.PCRCharacter;
import io.github.starwishsama.namelessbot.objects.group.GroupShop;
import io.github.starwishsama.namelessbot.objects.user.BotUser;
import io.github.starwishsama.namelessbot.objects.user.ClockInData;
import io.github.starwishsama.namelessbot.objects.user.RandomResult;

import java.util.*;
import java.util.concurrent.ConcurrentHashMap;

public class BotConstants {
    public static List<GroupShop> shop = new LinkedList<>();
    public static Collection<BotUser> users = new HashSet<>();
    public static List<BotLocalization> msg = new ArrayList<>();
    public static Config cfg = new Config();
    public static List<RandomResult> underCovers = new LinkedList<>();
    public static Map<Long, ClockInData> data = new HashMap<>();
    public static Map<String, Integer> repeatData = new ConcurrentHashMap<>();
    public static List<ArkNightOperator> operators = new LinkedList<>();
    public static List<PCRCharacter> pcr = new LinkedList<>();
}

File: src/main/kotlin/io/github/starwishsama/nbot/BotConstants.kt
package io.github.starwishsama.nbot

import com.google.gson.Gson
import com.google.gson.GsonBuilder
import com.google.gson.JsonObject
import io.github.starwishsama.nbot.objects.BotLocalization
import io.github.starwishsama.nbot.objects.BotUser
import io.github.starwishsama.nbot.objects.Config
import io.github.starwishsama.nbot.objects.RandomResult
import io.github.starwishsama.nbot.objects.draw.ArkNightOperator
import io.github.starwishsama.nbot.objects.draw.PCRCharacter
import io.github.starwishsama.nbot.objects.group.Shop
import java.time.format.DateTimeFormatter
import java.util.*


/**
 * 机器人(几乎)所有数据的存放类
 * 可以直接访问数据
 * @author Nameless
 */

object BotConstants {
    var shop: List<Shop> = LinkedList()
    var users: List<BotUser> = LinkedList()
    var msg: List<BotLocalization> = ArrayList()
    var cfg = Config()
    var underCovers: List<RandomResult> = LinkedList()
    var cache: JsonObject = JsonObject()

    /** 舟游/PCR 数据 */
    var arkNight: List<ArkNightOperator> = LinkedList()
    var pcr: List<PCRCharacter> = LinkedList()

    var gson: Gson = GsonBuilder().serializeNulls().setPrettyPrinting().create()
    val dateFormatter: DateTimeFormatter = DateTimeFormatter.ofPattern("HH:mm")
}